### PR TITLE
Create `AnalysisWorkflow` and `FeedbackAnalysisWorkflow` entities

### PR DESCRIPTION
**Pull request by rashidovasadbek**
close: #68

### CODE REVIEW

File: src/Feedback.Analyzer.Backend/Feedback.Analyzer.Domain/Enums/WorkflowType.cs [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+    /// <summary>
+    /// Indicates a training workflow.
+    /// </summary>
+    Training,
+
+    /// <summary>
+    /// Indicates an interface workflow.
+    /// </summary>
+    Interface
```
Comment (sulton-max) [inline src/Feedback.Analyzer.Backend/Feedback.Analyzer.Domain/Enums/WorkflowType.cs:21]: Change to `Inference`

File: src/Feedback.Analyzer.Backend/Feedback.Analyzer.Persistence/EntityConfiguration/AnalysisWorkflowConfiguration.cs [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+using Microsoft.EntityFrameworkCore;
+using Microsoft.EntityFrameworkCore.Metadata.Builders;
+
+namespace Feedback.Analyzer.Persistence.EntityConfiguration;
+
+public class AnalysisWorkflowConfiguration : IEntityTypeConfiguration<AnalysisWorkflow>
+{
+    public void Configure(EntityTypeBuilder<AnalysisWorkflow> builder)
+    {
+        builder.Property(analysisWorkflow => analysisWorkflow.Name).HasMaxLength(128).IsRequired();
```
Comment (sulton-max) [inline src/Feedback.Analyzer.Backend/Feedback.Analyzer.Persistence/EntityConfiguration/AnalysisWorkflowConfiguration.cs:11]: Add conversion for `Type` column and add length constraint

File: src/Feedback.Analyzer.Backend/Feedback.Analyzer.Persistence/Repositories/AnalysisWorkflowRepository.cs [hunk 3]
```diff
@@ -0,0 +1,29 @@
+using System.Linq.Expressions;
+using Feedback.Analyzer.Domain.Common.Commands;
+using Feedback.Analyzer.Domain.Common.Queries;
+using Feedback.Analyzer.Domain.Entities;
+using Feedback.Analyzer.Persistence.DataContexts;
+using Feedback.Analyzer.Persistence.Repositories.Interfaces;
+
+namespace Feedback.Analyzer.Persistence.Repositories;
+
+public class AnalysisWorkflowRepository(AppDbContext appDbContext) : EntityRepositoryBase<AnalysisWorkflow, AppDbContext>(appDbContext), IAnalysisWorkflowRepository
```
Comment (sulton-max) [inline src/Feedback.Analyzer.Backend/Feedback.Analyzer.Persistence/Repositories/AnalysisWorkflowRepository.cs:10]: Add type docs

**Review by sulton-max — CHANGES_REQUESTED** [review]

File: src/Feedback.Analyzer.Backend/Feedback.Analyzer.Persistence/EntityConfiguration/AnalysisWorkflowConfiguration.cs (hunk 2 above)
Comment (sulton-max) [inline src/Feedback.Analyzer.Backend/Feedback.Analyzer.Persistence/EntityConfiguration/AnalysisWorkflowConfiguration.cs:11]: Add type conversion and constraint

**Review by sulton-max — CHANGES_REQUESTED** [review]

**Review by sulton-max — CHANGES_REQUESTED** [review]

**Review by sulton-max — DISMISSED** [review]
Good job 👍👏

**Review by isxoqovxasanboy — APPROVED** [review]

File: src/Feedback.Analyzer.Backend/Feedback.Analyzer.Persistence/EntityConfiguration/AnalysisWorkflowConfiguration.cs [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+namespace Feedback.Analyzer.Persistence.EntityConfiguration;
+
+public class AnalysisWorkflowConfiguration : IEntityTypeConfiguration<AnalysisWorkflow>
+{
+    public void Configure(EntityTypeBuilder<AnalysisWorkflow> builder)
+    {
+        builder.Property(analysisWorkflow => analysisWorkflow.Name)
+            .HasMaxLength(128)
+        
+            .IsRequired();
```
Comment (botirjonabdumannonovv) [inline src/Feedback.Analyzer.Backend/Feedback.Analyzer.Persistence/EntityConfiguration/AnalysisWorkflowConfiguration.cs:14]: remove unnecessary row

**Review by sulton-max — APPROVED** [review]

**Review by botirjonabdumannonovv — APPROVED** [review]

File: src/Feedback.Analyzer.Backend/Feedback.Analyzer.Domain/Entities/AnalysisWorkflow.cs [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+    /// Gets or sets the name of the analysis workflow.
+    /// </summary>
+    public string Name { get; set; } = default!;
+
+    /// <summary>
+    /// Gets or sets the type of the analysis workflow.
+    /// </summary>
+    public WorkflowType Type { get; set; }
+
+    public FeedbackAnalysisWorkflow FeedbackAnalysisWorkflow = default!;
```
Comment (sulton-max) [inline src/Feedback.Analyzer.Backend/Feedback.Analyzer.Domain/Entities/AnalysisWorkflow.cs:21]: Remove navigation

File: src/Feedback.Analyzer.Backend/Feedback.Analyzer.Domain/Entities/FeedbackAnalysisWorkflow.cs [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+    public string Name { get; set; } = default!;
+    
+    /// <summary>
+    /// Gets or sets the type of the feedback analysis workflow.
+    /// </summary>
+    public WorkflowType Type { get; set; }
+
+    public AnalysisWorkflow AnalysisWorkflow { get; set; } = default!;
+
+    public Product Product { get; set; } = default!;
```
Comment (sulton-max) [inline src/Feedback.Analyzer.Backend/Feedback.Analyzer.Domain/Entities/FeedbackAnalysisWorkflow.cs:28]: Add docs

File: src/Feedback.Analyzer.Backend/Feedback.Analyzer.Api/Controllers/AnalysisWorkflowController.cs [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+        return Ok(result);
+    }
+
+    [HttpDelete("{analysisWorkflowId:guid}")]
+    public async ValueTask<IActionResult> DeleteAnalysisWorkflowById([FromRoute] Guid analysisWorkflowId, CancellationToken cancellationToken = default)
+    {
+        var result = await mediator.Send(new AnalysisWorkflowDeleteByIdCommand {AnalysisWorkflowId = analysisWorkflowId}, cancellationToken);
+        return result ? Ok() : BadRequest();
+    }
+ }
```
Comment (sulton-max) [inline src/Feedback.Analyzer.Backend/Feedback.Analyzer.Api/Controllers/AnalysisWorkflowController.cs:46]: Remove this controller

File: src/Feedback.Analyzer.Backend/Feedback.Analyzer.Application/AnalysisWorkflows/Commands/AnalysisWorkflowCreateCommand.cs [hunk 8]
```diff
@@ -0,0 +1,16 @@
...
+/// <summary>
+/// Represents a command for creating an analysis workflow.
+/// </summary>
+public record AnalysisWorkflowCreateCommand : ICommand<AnalysisWorkflowDto>
+{
+    /// <summary>
+    /// Gets or sets the analysis workflow data to create.
+    /// </summary>
+    public AnalysisWorkflowDto AnalysisWorkflow { get; set; } = default!;
+}
```
Comment (sulton-max) [inline src/Feedback.Analyzer.Backend/Feedback.Analyzer.Application/AnalysisWorkflows/Commands/AnalysisWorkflowCreateCommand.cs:16]: Remove

File: src/Feedback.Analyzer.Backend/Feedback.Analyzer.Application/AnalysisWorkflows/Commands/AnalysisWorkflowDeleteByIdCommand.cs [hunk 9]
```diff
@@ -0,0 +1,16 @@
...
+/// <summary>
+/// Represents a command for deleting an analysis workflow.
+/// </summary>
+public record AnalysisWorkflowDeleteByIdCommand : ICommand<bool>
+{
+    /// <summary>
+    /// The unique identifier of the Analysis Workflow to be deleted.
+    /// </summary>
+    public Guid AnalysisWorkflowId { get; set; }
+}
```
Comment (sulton-max) [inline src/Feedback.Analyzer.Backend/Feedback.Analyzer.Application/AnalysisWorkflows/Commands/AnalysisWorkflowDeleteByIdCommand.cs:16]: Remove

File: src/Feedback.Analyzer.Backend/Feedback.Analyzer.Application/AnalysisWorkflows/Commands/AnalysisWorkflowUpdateCommand.cs [hunk 10]
```diff
@@ -0,0 +1,15 @@
...
+/// <summary>
+/// Represents a command for updating an analysis workflow.
+/// </summary>
+public record AnalysisWorkflowUpdateCommand : ICommand<AnalysisWorkflowDto>
+{
+    /// <summary>
+    /// Gets or sets the analysis workflow data to update.
+    /// </summary>
+    public AnalysisWorkflowDto AnalysisWorkflow { get; set; } = default!;
+}
```
Comment (sulton-max) [inline src/Feedback.Analyzer.Backend/Feedback.Analyzer.Application/AnalysisWorkflows/Commands/AnalysisWorkflowUpdateCommand.cs:15]: Remove

File: src/Feedback.Analyzer.Backend/Feedback.Analyzer.Application/AnalysisWorkflows/Models/AnalysisWorkflowDto.cs [hunk 11]
```diff
@@ -0,0 +1,24 @@
...
+    /// <summary>
+    /// Gets or sets the name of the analysis workflow.
+    /// </summary>
+    public string Name { get; set; } = default!;
+
+    /// <summary>
+    /// Gets or sets the type of the analysis workflow.
+    /// </summary>
+    public WorkflowType Type { get; set; }
+}
```
Comment (sulton-max) [inline src/Feedback.Analyzer.Backend/Feedback.Analyzer.Application/AnalysisWorkflows/Models/AnalysisWorkflowDto.cs:24]: Remove

File: src/Feedback.Analyzer.Backend/Feedback.Analyzer.Application/AnalysisWorkflows/Queries/AnalysisWorkflowGetByIdQuery.cs [hunk 12]
```diff
@@ -0,0 +1,15 @@
...
+/// <summary>
+/// Represents a query for retrieving an analysis workflow by its ID.
+/// </summary>
+public class AnalysisWorkflowGetByIdQuery : IQuery<AnalysisWorkflowDto>
+{
+    /// <summary>
+    /// Gets or sets the ID of the analysis workflow to retrieve.
+    /// </summary>
+    public Guid AnalysisWorkflowId { get; set; } = Guid.Empty;
+}
```
Comment (sulton-max) [inline src/Feedback.Analyzer.Backend/Feedback.Analyzer.Application/AnalysisWorkflows/Queries/AnalysisWorkflowGetByIdQuery.cs:15]: Remove

File: src/Feedback.Analyzer.Backend/Feedback.Analyzer.Application/AnalysisWorkflows/Queries/AnalysisWorkflowGetQuery.cs [hunk 13]
```diff
@@ -0,0 +1,15 @@
...
+/// <summary>
+/// Represents a query for retrieving a collection of analysis workflows based on a filter.
+/// </summary>
+public class AnalysisWorkflowGetQuery : IQuery<ICollection<AnalysisWorkflowDto>>
+{
+    /// <summary>
+    /// Gets or sets the filter to apply when retrieving analysis workflows.
+    /// </summary>
+    public AnalysisWorkflowFilter Filter { get; set; } = default!;
+}
```
Comment (sulton-max) [inline src/Feedback.Analyzer.Backend/Feedback.Analyzer.Application/AnalysisWorkflows/Queries/AnalysisWorkflowGetQuery.cs:15]: Remove

File: src/Feedback.Analyzer.Backend/Feedback.Analyzer.Infrastructure/AnalysisWorkflows/CommandHandlers/AnalysisWorkflowCreateCommandHandler.cs [hunk 14]
```diff
@@ -0,0 +1,30 @@
...
+        
+        // call service
+        var createAnalysisWorkflow = await analysisWorkflowService.CreateAsync(analysisWorkflow, cancellationToken: cancellationToken);
+        
+        // Conversation to Dto
+        var analysisWorkflowDto = mapper.Map<AnalysisWorkflowDto>(createAnalysisWorkflow);
+
+        return analysisWorkflowDto;
+    }
+}
```
Comment (sulton-max) [inline src/Feedback.Analyzer.Backend/Feedback.Analyzer.Infrastructure/AnalysisWorkflows/CommandHandlers/AnalysisWorkflowCreateCommandHandler.cs:30]: Remove

File: src/Feedback.Analyzer.Backend/Feedback.Analyzer.Infrastructure/AnalysisWorkflows/CommandHandlers/AnalysisWorkflowDeleteByIdCommandHandler.cs [hunk 15]
```diff
@@ -0,0 +1,22 @@
...
+{
+  public  async Task<bool> Handle(AnalysisWorkflowDeleteByIdCommand request, CancellationToken cancellationToken)
+  {
+      // Delete the analysis workflow by its ID using the service
+      await analysisWorkflowService.DeleteByIdAsync(request.AnalysisWorkflowId, cancellationToken: cancellationToken);
+      
+      // Return true indicating the deletion was successful
+      return true;
+  }
+}
```
Comment (sulton-max) [inline src/Feedback.Analyzer.Backend/Feedback.Analyzer.Infrastructure/AnalysisWorkflows/CommandHandlers/AnalysisWorkflowDeleteByIdCommandHandler.cs:22]: Remove

File: src/Feedback.Analyzer.Backend/Feedback.Analyzer.Infrastructure/AnalysisWorkflows/CommandHandlers/AnalysisWorkflowUpdateCommandHandler.cs [hunk 16]
```diff
@@ -0,0 +1,29 @@
...
+        
+        // call service
+        var createAnalysisWorkflow = await analysisWorkflowService.UpdateAsync(analysisWorkflow, cancellationToken: cancellationToken);
+        
+        // Conversation to Dto
+        var analysisWorkflowDto = mapper.Map<AnalysisWorkflowDto>(createAnalysisWorkflow);
+
+        return analysisWorkflowDto;
+    }
+}
```
Comment (sulton-max) [inline src/Feedback.Analyzer.Backend/Feedback.Analyzer.Infrastructure/AnalysisWorkflows/CommandHandlers/AnalysisWorkflowUpdateCommandHandler.cs:29]: Remove

File: src/Feedback.Analyzer.Backend/Feedback.Analyzer.Infrastructure/AnalysisWorkflows/Mappers/AnalysisWorkflowMapper.cs [hunk 17]
```diff
@@ -0,0 +1,13 @@
...
+
+namespace Feedback.Analyzer.Infrastructure.AnalysisWorkflows.Mappers;
+
+public class AnalysisWorkflowMapper : Profile
+{
+    public AnalysisWorkflowMapper()
+    {
+        CreateMap<AnalysisWorkflow, AnalysisWorkflowDto>().ReverseMap();
+    }
+}
```
Comment (sulton-max) [inline src/Feedback.Analyzer.Backend/Feedback.Analyzer.Infrastructure/AnalysisWorkflows/Mappers/AnalysisWorkflowMapper.cs:13]: Remove

File: src/Feedback.Analyzer.Backend/Feedback.Analyzer.Infrastructure/AnalysisWorkflows/QueryHandlers/AnalysisWorkflowGetByIdQueryHandler.cs [hunk 18]
```diff
@@ -0,0 +1,26 @@
...
+        var foundAnalysisWorkflow = await analysisWorkflowService.GetByIdAsync(
+            request.AnalysisWorkflowId,
+            new QueryOptions
+            {
+                AsNoTracking = true
+            }, cancellationToken);
+
+        return mapper.Map<AnalysisWorkflowDto>(foundAnalysisWorkflow);
+    }
+}
```
Comment (sulton-max) [inline src/Feedback.Analyzer.Backend/Feedback.Analyzer.Infrastructure/AnalysisWorkflows/QueryHandlers/AnalysisWorkflowGetByIdQueryHandler.cs:26]: Remove

File: src/Feedback.Analyzer.Backend/Feedback.Analyzer.Infrastructure/AnalysisWorkflows/QueryHandlers/AnalysisWorkflowGetQuery.cs [hunk 19]
```diff
@@ -0,0 +1,23 @@
...
+    : IQueryHandler<AnalysisWorkflowGetQuery, ICollection<AnalysisWorkflowDto>>
+{
+    public async Task<ICollection<AnalysisWorkflowDto>> Handle(AnalysisWorkflowGetQuery request, CancellationToken cancellationToken)
+    {
+        var matchedAnalysisWorkflow = await analysisWorkflowService
+            .Get(request.Filter, new QueryOptions { AsNoTracking = true }).ToListAsync(cancellationToken);
+
+        return mapper.Map<ICollection<AnalysisWorkflowDto>>(matchedAnalysisWorkflow);
+    }
+}
```
Comment (sulton-max) [inline src/Feedback.Analyzer.Backend/Feedback.Analyzer.Infrastructure/AnalysisWorkflows/QueryHandlers/AnalysisWorkflowGetQuery.cs:23]: Remove

File: src/Feedback.Analyzer.Backend/Feedback.Analyzer.Persistence/EntityConfiguration/FeedbackAnalysisWorkflowConfiguration.cs [hunk 20]
```diff
@@ -0,0 +1,34 @@
...
+            .HasMaxLength(128)
+            .IsRequired();
+        
+        builder.HasIndex(
+            workflow => new
+            {
+                workflow.Name,
+                workflow.ProductId
+            }
+        ).IsUnique();
```
Comment (sulton-max) [inline src/Feedback.Analyzer.Backend/Feedback.Analyzer.Persistence/EntityConfiguration/FeedbackAnalysisWorkflowConfiguration.cs:27]: Remove this and add 

"TODO : Add name and product Id unique index between this and `AnalysisWorkflow` tables"

File: src/Feedback.Analyzer.Backend/Feedback.Analyzer.Persistence/Repositories/Interfaces/IFeedbackAnalysisWorkflowRepository.cs [hunk 21]
```diff
@@ -0,0 +1,62 @@
+using System.Linq.Expressions;
+using Feedback.Analyzer.Domain.Common.Commands;
+using Feedback.Analyzer.Domain.Common.Queries;
+using Feedback.Analyzer.Domain.Entities;
+
+namespace Feedback.Analyzer.Persistence.Repositories.Interfaces;
+
+public interface IFeedbackAnalysisWorkflowRepository
```
Comment (sulton-max) [inline src/Feedback.Analyzer.Backend/Feedback.Analyzer.Persistence/Repositories/Interfaces/IFeedbackAnalysisWorkflowRepository.cs:8]: Add type docs

File: src/Feedback.Analyzer.Backend/Feedback.Analyzer.Persistence/Repositories/FeedbackAnalysisWorkflowRepository.cs [hunk 22]
```diff
@@ -0,0 +1,46 @@
+using System.Linq.Expressions;
+using Feedback.Analyzer.Domain.Common.Commands;
+using Feedback.Analyzer.Domain.Common.Queries;
+using Feedback.Analyzer.Domain.Entities;
+using Feedback.Analyzer.Persistence.DataContexts;
+using Feedback.Analyzer.Persistence.Repositories.Interfaces;
+
+namespace Feedback.Analyzer.Persistence.Repositories;
+
+public class FeedbackAnalysisWorkflowRepository(AppDbContext appDbContext)
```
Comment (sulton-max) [inline src/Feedback.Analyzer.Backend/Feedback.Analyzer.Persistence/Repositories/FeedbackAnalysisWorkflowRepository.cs:10]: Add type docs

File: src/Feedback.Analyzer.Backend/Feedback.Analyzer.Persistence/Migrations/20240308062843_Add FeedbackAnalysisWorkflow_And_AnalysisWorkflow.cs [hunk 23]
```diff
@@ -0,0 +1,47 @@
...
+    {
+        /// <inheritdoc />
+        protected override void Up(MigrationBuilder migrationBuilder)
+        {
+            migrationBuilder.CreateTable(
+                name: "FeedbackAnalysisWorkflows",
+                columns: table => new
+                {
+                    Id = table.Column<Guid>(type: "uuid", nullable: false),
+                    Name = table.Column<string>(type: "character varying(128)", maxLength: 128, nullable: false),
```
Comment (sulton-max) [inline src/Feedback.Analyzer.Backend/Feedback.Analyzer.Persistence/Migrations/20240308062843_Add FeedbackAnalysisWorkflow_And_AnalysisWorkflow.cs:19]: Remove name prop

File: src/Feedback.Analyzer.Backend/Feedback.Analyzer.Persistence/Migrations/20240308062843_Add FeedbackAnalysisWorkflow_And_AnalysisWorkflow.cs [hunk 24]
```diff
@@ -0,0 +1,47 @@
...
+                    Type = table.Column<string>(type: "character varying(128)", maxLength: 128, nullable: false)
```
Comment (sulton-max) [inline src/Feedback.Analyzer.Backend/Feedback.Analyzer.Persistence/Migrations/20240308062843_Add FeedbackAnalysisWorkflow_And_AnalysisWorkflow.cs:20]: Remove type prop

**Review by sulton-max — CHANGES_REQUESTED** [review]

**Review by sulton-max — APPROVED** [review]
Good job 👏

**Review by botirjonabdumannonovv — APPROVED** [review]

**Review by isxoqovxasanboy — APPROVED** [review]

**Pull request merged**